File: lang/ruby/src/vac.fun/algorithms/number_theory/LeastCommonMultiple.rb
require "vac.fun/algorithms/number_theory/GreatestCommonDivisor"

module LeastCommonMultiple

    def self.reduceToBinaryGCD(m, n)
        if m.negative?
            m = -m
        end
        if n.negative?
            n = -n
        end

        if m.zero? || n.zero?
            return 0
        end
        m / GreatestCommonDivisor.iterativeBinaryGCD(m, n) * n
    end

    def self.reduceToEuclidean(m, n)
        if m.negative?
            m = -m
        end
        if n.negative?
            n = -n
        end

        if m.zero? || n.zero?
            return 0
        end
        m / GreatestCommonDivisor.iterativeEuclidean(m, n) * n
    end
end

File: lang/nodejs/src/vac.fun/algorithms/number_theory/LeastCommonMultiple.js
"use strict"

const GreatestCommonDivisor = require("vac.fun/algorithms/number_theory/GreatestCommonDivisor")

exports.reduceToBinaryGCD = (m, n) => {
    if (m < 0) {
        m = -m
    }
    if (n < 0) {
        n = -n
    }

    if (m === 0 || n === 0) {
        return 0
    }
    return m / GreatestCommonDivisor.iterativeBinaryGCD(m, n) * n
}

exports.reduceToEuclidean = (m, n) => {
    if (m < 0) {
        m = -m
    }
    if (n < 0) {
        n = -n
    }

    if (m === 0 || n === 0) {
        return 0
    }
    return m / GreatestCommonDivisor.iterativeEuclidean(m, n) * n
}
